### PR TITLE
Fix/stacking copy updates

### PR DESCRIPTION
**Pull request by kyranjamie**
> [Download the latest build](https://github.com/blockstack/stacks-wallet/actions/runs/806672320)

### Todo
- [x] Add warning when stacking near threshold
- [x] Annotate all external links with arrows, disclaimer

### CODE REVIEW

**Comment by markmhx** [comment]
The "This entered amount would get you N reward slots" subcopy doesn't appear to change as expected upon change of entered value. This paragraph should be hidden when there's no entered value and updated with the accurate rewards slots estimate as entered or changed.

Screen Shot 2021-04-28 at 16 04 30


Screen Shot 2021-04-28 at 16 04 34

**Comment by markmhx** [comment]
A space is needed after the period for the first sentence here: 

image

**Comment by markmhx** [comment]
This subcopy needs updating:

image

It should be:

image

**Comment by markmhx** [comment]
We need to add the confirmation checkbox, which is currently missing:

image

**Comment by markmhx** [comment]
The Stacking guide card isn't linked to anywhere and shows a selection cursor on hover:

image

**Comment by markmhx** [comment]
Let's remove the trailing period from here:

image

**Comment by markmhx** [comment]
This bullet on the pooling screen needs updating generally:

image

Currently it looks like:

image

**Comment by markmhx** [comment]
This final bullet on the pooling screen also needs its subheader updated:

image

Currently it looks like:

image

**Comment by markmhx** [comment]
Aside from the above comments, we need to add [the dynamic minimum warning](https://github.com/blockstack/stacks-wallet/issues/693) and hover tooltip for all external links marked with an arrow (↗) with the following text:

> Hiro PBC provides no guarantees regarding this 3rd party website

cc @fjl4 

Otherwise, looks good for inclusion in https://github.com/blockstack/stacks-wallet/pull/718

**Comment by fjl4** [comment]
> Aside from the above comments, we need to add [the dynamic minimum warning](https://github.com/blockstack/stacks-wallet/issues/693) and hover tooltip for all external links marked with an arrow (↗) with the following text:
> 
> > Hiro PBC provides no guarantees regarding this 3rd party website
> 

Is it alright if we have the external link warning read:

"This link will take you to an external third-party website that is not affiliated with Hiro Systems PBC."

File: app/constants/index.ts [hunk 1]
```diff
@@ -41,6 +41,12 @@ export const TREZOR_HELP_URL =
 export const STACKING_ADDRESS_FORMAT_HELP_URL =
   'https://www.hiro.so/questions/what-form-of-btc-addresses-can-i-use-for-proof-of-transfer-rewards';
 
+export const STACKING_LEARN_MORE_URL = 'https://stacks.org/stacking-moves-us-forward';
+
+export const STACKING_GUIDE_URL = 'https://blog.stacks.co/stacking-strategy';
+
+export const STACKING_MINIMIUM_FOR_NEXT_CYCLE_URL = 'https://stacking.club/cycles/next';
```
Comment (aulneau) [inline app/constants/index.ts:48]:
```suggestion
export const STACKING_MINIMIUM_FOR_NEXT_CYCLE_URL = 'https://stacking.club/cycles/next?tab=slot_minimum';
```

File: app/pages/stacking/components/stacking-form-step.tsx [hunk 2]
```diff
@@ -73,9 +74,9 @@ export const StackingStep: FC<StackingFormStepProps> = props => {
 };
 
 export const StackingStepDescription: FC = ({ children }) => (
-  <Text display="block" textStyle="body.large" mt="tight">
+  <Stack display="block" textStyle="body.large" mt="tight" spacing="base">
```
Comment (aulneau) [inline app/pages/stacking/components/stacking-form-step.tsx:77]: rm `mt="tight"`

File: app/pages/stacking/components/stacking-guide-card.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+import { STACKING_GUIDE_URL } from '@constants/index';
+
+import { StackingFormInfoCard } from './stacking-form-info-card';
+import { LegalDisclaimerTooltip } from '@components/legal-disclaimer-tooltip';
+
+type StackingGuideCardProps = FlexProps;
+
+const openStackingGuide = () => openExternalLink(STACKING_GUIDE_URL);
+
+export const StackingGuideCard: FC<StackingGuideCardProps> = props => {
```
Comment (aulneau) [inline app/pages/stacking/components/stacking-guide-card.tsx:16]:
```suggestion
const openStackingGuide = () => openExternalLink(STACKING_GUIDE_URL);

export const StackingGuideCard: FC<FlexProps> = props => {
```

File: app/pages/stacking/components/stacking-term.tsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+        <Icon width={['12px', '16px']} />
+      </Box>
+      <Box>
+        <Text as="h3" textStyle="body.large.medium">
+          {title}
+        </Text>
+        <Stack mt="extra-tight" spacing="base" textStyle="body.large" color={color('text-caption')}>
+          {children}
+        </Stack>
+      </Box>
```
Comment (aulneau) [inline app/pages/stacking/components/stacking-term.tsx:22]:
```suggestion
      <Stack spacing="extra-tight">
        <Text as="h3" textStyle="body.large.medium">
          {title}
        </Text>
        <Stack spacing="base" textStyle="body.large" color={color('text-caption')}>
          {children}
        </Stack>
      </Stack>
```

File: app/pages/stacking/direct-stacking/components/choose-btc-address.tsx [hunk 5]
```diff
@@ -56,6 +56,12 @@ export const ChooseBtcAddressStep: FC<ChooseBtcAddressStepProps> = props => {
   const form = useFormik({
     initialValues: { btcAddress: '' },
     validate: ({ btcAddress }) => {
+      const isValid = validate(btcAddress);
```
Comment (aulneau) [inline app/pages/stacking/direct-stacking/components/choose-btc-address.tsx:59]: name conflict?

File: app/pages/stacking/direct-stacking/components/direct-stacking-intro.tsx [hunk 6]
```diff
@@ -10,10 +10,14 @@ interface StackingIntroProps {
...
   <>
-    <StackingTitle>Start earning Bitcoin</StackingTitle>
+    <StackingTitle>Stack by yourself</StackingTitle>
     <StackingDescription mt="base-tight">
-      Help secure the Stacks blockchain and earn Bitcoin by temporarily locking your STX in your
-      wallet. Your STX will never leave your wallet.
+      When you stack by yourself, you’ll get the chance to earn Bitcoin each cycle for every reward
+      slot that you hold.
+      <br />
+      <br /> The STX required per reward slot can fluctuate from cycle to cycle. If you’re close to
```
Comment (aulneau) [inline app/pages/stacking/direct-stacking/components/direct-stacking-intro.tsx:18]: would be better to do two `<Box>` and have margin than rely on line-breaks imo

File: app/pages/stacking/direct-stacking/components/choose-btc-address.tsx (hunk 5 above)
Comment (kyranjamie) [reply app/pages/stacking/direct-stacking/components/choose-btc-address.tsx:59]: No, it's fine as the other validate is only an object prop name

File: app/assets/images/abstract-btc-tower.svg [hunk 7]
```diff
@@ -0,0 +1,21 @@
+<svg width="56" height="56" viewBox="0 0 56 56" fill="none" xmlns="http://www.w3.org/2000/svg">
```
[the remark this answers is not in the document]
Comment (kyranjamie) [reply app/assets/images/abstract-btc-tower.svg:1]: image

As this is more of an illustration than an adaptable icon, I don't think this is necessary.

**Comment by markmhx** [comment]
I've been struggling with poor WiFi during travel this weekend (and unable to download the latest build) but will prioritize review of previously requested changes on Monday / tomorrow once back at home.

**Comment by markmhx** [comment]
The buffer value here in the copy doesn't appear to change e.g. when I increase from the minimum to the minimum plus 1 STX:

Screen Shot 2021-05-03 at 13 05 06
Screen Shot 2021-05-03 at 13 05 14

**Comment by markmhx** [comment]
These updates look good generally other than the one above comment regarding the buffer value 👍

I'm thinking on second glance that https://github.com/blockstack/stacks-wallet/issues/693 may not be necessary given we have this dynamic copy below the amount field in the direct Stacking field. That warning element doesn't appear integrated yet in any case. @jasperjansz @GinaAbrams do you think it's needed given how much we emphasize the buffer and context here already in this build?

image

**Review by markmhx — CHANGES_REQUESTED** [review]
Buffer value in copy needs dynamic updating.

**Comment by kyranjamie** [comment]
Fixed pluralisation of reward slots. @markmhx is this necessary to change before releasing? The copy above the warning explains a reward slot is 10k.

**Comment by markmhx** [comment]
We can release if you prefer. I just suspect that users may not equate reward slot with displayed buffer value without further indication.

**Pull request merged**